bpf/bpf_host.c

```diff
@@ -365,13 +365,11 @@ handle_ipv6(struct __ctx_buff *ctx, __u32 secctx, const bool from_host)
 	if (hdrlen < 0)
 		return hdrlen;
 
-#ifdef HANDLE_NS
```

Comment:
I'm not sure I buy the argument on this one.

The reason we handle the NS for the pods case is that Cilium is acting as an L3 router between the pods and the external network. We terminate the L2 network at each pod so they all see a tiny L2 with only themselves and the Cilium router.

When we insert such logic between the external network and the host itself, then Cilium is becoming a router between the network and the host? I don't fully understand the ramifications here.

Reply:
(Incidentally, I note that this changes IPv6 path but the equivalent IPv4 path should be considered with the same arguments).

Reply:
I didn't change this. `HANDLE_NS` was always set to 1 (in `init.sh`) before I introduced the host endpoint so this snippet of code was always executed.

We answer to NS and echo requests here only. I'm a bit wary of removing this code without understanding why it was implemented there in the first place. It dates to the very beginning of Cilium so maybe @tgraf knows?

Reply:
I dug a little deeper here, specifically in `__icmp6_handle_ns()` and it seems like basically we are replying specifically for ND to the router IP, and dropping all other ND. My understanding is that this would apply for traffic coming into the stack on the cilium veth pair when running Cilium 1.7 or earlier. Maybe there's a chaining argument here where directly connected non-Cilium networks (eg docker bridge) could have hosts hanging off which could solicit the Cilium router IP, in which case we should respond...? I'm struggling to visualize where such requests might actually come from in such a way that they would hit the cilium veth pair.

One of my points here is that with host firewall + multi-dev support we're now extending this across the native devices as well as the cilium veth pair, which is new behaviour in this upcoming release. 

Would still be interested if @tgraf has further insights.

Reply:
Thomas answered on Slack:
> The idea was the make the virtual Cilium router pingable

We should have the same behavior if we just send ICMPv6 packets up the stack.